src/ledger.js

```diff
@@ -170,12 +169,26 @@ export class LedgerInteraction extends DirectKeystoreInteraction {
       const transport = await TransportWebUsb.create();
       return callback(transport);
     } catch (e) {
-      if (e.message && e.message === "navigator.usb is undefined") { // webusb failed, so try u2f
-        const transport = await TransportU2F.create();
-        return callback(transport);
-      } else {
-        throw new Error("Error with Ledger transport");
+      if (e.message) {
+        //WebUSB Errors:
+        // If you hit cancel on the browser webusb popup
+        if (e.message === 'No device selected.') {
+          e.message = `Select your device in the WebUSB dialog box. Make sure it's plugged in, unlocked, and has the Bitcoin app open.`;
+        }
+        if (e.message === 'undefined is not an object (evaluating \'navigator.usb.getDevices\')') {
+          e.message = `Safari is not a supported browser.`;
+        }
+        // If you ask for a Transport in Firefox, fallback to U2F
+        if (e.message === "navigator.usb is undefined") {
+          try {
+            const transport = await TransportU2F.create();
```

Comment:
We have to make this switch in error handling because of the nano x, right? I'm asking to make sure it's not something we can decide by passing in additional params at instantiation.

Reply:
All of this was a hack to switch the Transport we call from webusb over to the U2F transport in the case of a Ledger device interaction request from Firefox. The Nano X does not show up in the device list of webusb via Chrome on Windows for ... 'complicated firmware reasons'. Therefore, Nano X on Windows would be unable to do anything without this fallback.

It's just a bonus that Nano S/X work in FF on other OSes with this change.

Reply:
Yeah, because nano x doesn't work in Chrome on Windows 10 version 1909 but works in Windows 10 version 2004 (at least on my VM).

What Nikki is suggesting is cleaner but I'm not sure how difficult it would be.  If it's difficult then I'd personally be okay with this approach as u2f's days are numbered anyway.